### ISSUE TITLE
artifactory_file checksum issue

### ISSUE THREAD

**Issue by sixxta** [jfrog/terraform-provider-artifactory#441]
**Describe the bug**
I'm trying to create a POC of the Artifactory provider to then use to refactor our internal `lambda` module. 

```
terraform {
  required_providers {
    artifactory = {
      source  = "registry.terraform.io/jfrog/artifactory"
      version = "6.6.0"
    }
  }
}

provider "artifactory" {
  url           = "http://host.blah.blah"
  api_key       = "[redacted]"
}

data "artifactory_file" "my-file" {
  repository  = "cloud"
  path        = "log-renderer-17.zip"
  output_path = "log-renderer-17.zip"
}

output "name" {
  value = data.artifactory_file.my-file
}
```

I can curl the zip file locally no problem (at `http://host.blah.blah/artifactory/cloud/log-renderer-17.zip`, along with the API endpoint I see in the code [here](https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/datasource/datasource_artifactory_file.go#L140)

When I run a plan on the above TF, I get the following:
```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.artifactory_file.my-file: Refreshing state...

Error: log-renderer-17.zip checksum and  checksum do not match, expectd 

  on main.tf line 34, in data "artifactory_file" "my-file":
  34: data "artifactory_file" "my-file" {
```

This makes it seem like the `fileInfo` struct does not get filled in. It does download a file into the `output_path` that seems to be a html page. 

Artifactory version: EnterpriseX license 7.33.9 rev
Terraform version: 0.13.7
Terraform provider: 6.6.0 (but also tried 6.5.2 and 6.5.3)

**Requirements for and issue**
- [X] A description of the bug
- [X] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). 
- [X] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [X] Your version of terraform
- [X] Your version of terraform provider

**Expected behavior**
File is downloaded from Artifactory, `output` is populated with correct values and then can be used in `lambda` module.

**Comment by alexhung** [comment]
@sixxta Thanks for the report. We'll investigate. There were a few similar reports a few weeks ago but we could not reproduce. They were all lambda related so can I ask you to execute the Terraform plan locally and see if this error still happens?

**Comment by sixxta** [comment]
Yup, still get the same thing upon an `apply`

**Comment by chb0github** [comment]
@alexhung - it might make sense to output some logging - if there hashes don't line up I'd love to know why - I am starting to think it's a bug in Artifactory

**Comment by alexhung** [comment]
@sixxta We've included additional debug logging in released [v6.8.1](https://registry.terraform.io/providers/jfrog/artifactory/6.8.1).

Can you test again and capture the logs for us?

Steps:
- Delete the file to be downloaded in the target location if exist (provider won't download it if already exists)
- Execute the `terraform apply` but now prefix with `TF_LOG=DEBUG`, e.g. `TF_LOG=DEBUG terraform apply`.
- You'll see *lots* of output in your shell. Copy them and share with us via public gist. Or if you feel more comfortable, email us at partner-support@jfrog.com

**Comment by chb0github** [comment]
If we don't get logs we can't proceed. We aren't able to repro this but we know this is a funky api. 
@sixxta - what type of repo are you upload this file into?

**Comment by sixxta** [comment]
Hi there @chb0github and @alexhung I sent all the relevant information to the email address, it'll be from emorales@fanatics.com. Thanks!

**Comment by alexhung** [comment]
@sixxta Just want to confirm that you've emailed to partner-support@jfrog.com. We haven't received any email so far.

**Comment by sixxta** [comment]
> 

Yup I just resent it, and the email address is correct.

**Comment by alexhung** [comment]
@sixxta Great! I'll check with IT to see if the email is flagged or something.

**Comment by alexhung** [comment]
@sixxta Our email system blocked the email due to the content. Would it be possible for you to share the logs in a private Gist?

**Comment by chb0github** [comment]
Ok, @sixxta 

Here is a [public upload link](https://drive.google.com/drive/folders/1_nAOfVhjXIIAJwM3UuiPdReLiFrhdC65?usp=sharing) where you can dump the logs. I understand we're all trying to sort this out, but if we can't get some logs soon, we really need to just close this ticket. Please do all you can to upload

**Comment by sixxta** [comment]
Here's the gist for you: https://gist.github.com/sixxta/cc19f939809072bc828c0add24b7b812

**Comment by alexhung** [comment]
@sixxta The log indicates that the file is already downloaded and exists on the target path:

> WARN-001: file download skipped. fileExists: true, forceOverwrite: false

So can you either delete the file or set `forceOverwrite` to `true` to force a download?

Let me know when you have the logs for this.

**Comment by sixxta** [comment]
Updated the gist with new output - under force-overwrite-output.txt

**Comment by alexhung** [comment]
@sixxta I took a look at the log and did not see any error downloading the file. Is this ran with HCL that causes the original issue?

**Comment by chb0github** [comment]
I looked as well. I don't see the issue.

**Comment by chb0github** [comment]
Please send us your license per the readme and we can debug that separately. I'll create a ticket once you do. 

What HTML do you get when you render it? Can you give us a screenshot? I mean, we have no control over what Artifactory gives you but if we see the HTML maybe it will clue us in.

**Comment by sixxta** [comment]
I made a curl the following endpoint `http://repo.lab.frgcloud.com/artifactory/api/system/licenses/` with username:token and got what's in the gist above as `licenses.txt`. As for the HTML, I got nothing when I render it, but the code can be seen as `log-renderer-17.zip` also in the gist.

Here's the link: https://gist.github.com/sixxta/cc19f939809072bc828c0add24b7b812

**Comment by chb0github** [comment]
I will discuss this with @alexhung and see how to proceed - What we really need is details about what it got from artifactory during API calls.

**Comment by chb0github** [comment]
We haven't been able to reproduce this locally, and without better logs (which, is on us as the provider currently doesn't log request/response and that's what is needed here) we really can't say what is going on. Though, I emphasize again that, if you're getting _a file at all_ then, from our perspective it seems to work (we have no control over what artifactory produces).

**Comment by alexhung** [comment]
@sixxta I took a look at the HTML content of the zip file and I noticed references to `/vpn`, and 'Citrix' (https://gist.github.com/sixxta/cc19f939809072bc828c0add24b7b812#file-log-renderer-17-zip-L53). Is the file stored on a local repository or remote? The HTML content suggests that the file path is protected by some kind of authentication (VPN? Citrix?) and Artifactory can only fetched the HTML page (instead of the actual zip file).

We will be adding [more extensive debug logging](https://github.com/jfrog/terraform-provider-shared/issues/16) so this should help future troubleshooting.
